packages/core/src/Site/index.js

```diff
@@ -856,8 +873,24 @@ class Site {
     const removedPages = _.differenceWith(oldAddressablePages, this.addressablePages, isNewPage)
       .map(filePath => Site.setExtension(filePath.src, '.html'));
 
-    if (!_.isEmpty(addedPages) || !_.isEmpty(removedPages)) {
+    // Checks if any attributes of site.json requiring a global rebuild are modified
+    const isGlobalConfigModified = () => !_.isEqual(oldSiteConfig.faviconPath, this.siteConfig.faviconPath)
+        || !_.isEqual(oldSiteConfig.titlePrefix, this.siteConfig.titlePrefix)
+        || !_.isEqual(oldSiteConfig.style, this.siteConfig.style)
+        || !_.isEqual(oldSiteConfig.externalScripts, this.siteConfig.externalScripts)
+        || !_.isEqual(oldSiteConfig.globalOverride, this.siteConfig.globalOverride)
+        || !_.isEqual(oldSiteConfig.plugins, this.siteConfig.plugins)
+        || !_.isEqual(oldSiteConfig.pluginsContext, this.siteConfig.pluginsContext)
+        || !_.isEqual(oldSiteConfig.headingIndexingLevel, this.siteConfig.headingIndexingLevel)
+        || !_.isEqual(oldSiteConfig.enableSearch, this.siteConfig.enableSearch)
+        || !_.isEqual(oldSiteConfig.disableHtmlBeautify, this.siteConfig.disableHtmlBeautify)
+        || !_.isEqual(oldSiteConfig.timeZone, this.siteConfig.timeZone)
+        || !_.isEqual(oldSiteConfig.locale, this.siteConfig.locale)
+        || !_.isEqual(oldSiteConfig.intrasiteLinkValidation, this.siteConfig.intrasiteLinkValidation);
+
+    if (isGlobalConfigModified() || !_.isEmpty(addedPages) || !_.isEmpty(removedPages)) {
       await this.removeAsset(removedPages);
+      this.buildManagers();
       await this._rebuildSourceFiles();
```

Comment:
Additional remark (no action needed): 

>> EDIT: After looking around at the other handlers (addHandler and removeHandler), it seems all pages should be rebuilt if the added/removed file is a page/dependency of page.
>
>This has been existing behaviour for a long while already but can probably be made into an issue 🙂

Per #1514's comment, the number of dependent pages generated / regenerated can be improved later as well.

Reply:
Alright. I'll open an issue for this and the previous discussion.